src/api/app.js

```diff
@@ -69,10 +69,7 @@ module.exports.authorise = (appToken, permissions, options) => {
 module.exports.connectAuthorised = (appToken, authUri) => {
   return getObj(appToken)
     .then((app) => app.auth.loginFromURI(authUri))
-    .then((connectedApp) => {
-      freeObj(appToken);
-      return genHandle(connectedApp);
-    });
+    .then((connectedApp) => appToken);
```

Comment:
@shankar2105 you've referenced the wrong piece of code. The code is calling https://github.com/maidsafe/safe_app_nodejs/blob/master/src/index.js#L58 -- which isn't creating any new instance of SAFEApp but only mutates the existing one (by replacing the connection to the remote Rust object -- see https://github.com/maidsafe/safe_app_nodejs/blob/master/src/app.js#L135 ).

The function you have referenced is/can only be used when there isn't any SAFEApp instance yet -- thus it only ever creates but never _replaces_ any existing instance.